python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1500,15 +1500,41 @@ def should_backfill_atomic_asset_partitions_unit(
                 f" {parent_partitions_result.required_but_nonexistent_parents_partitions}"
             )
 
+        asset_partitions_to_request_map: Dict[AssetKey, Set[str]] = {}
```

Comment:
`partition_key` on AssetKeyPartitionKey can be `None` (for unpartitioned assets), so this type signature should be `Dict[AssetKey, Set[Optional[str]]]`. You can run `make pyright` to confirm that the types line up after updating there.

Also, you can save yourself some pain later on if you just use `asset_partitions_to_request_map: ... = defaultdict(set)`. That way, you don't have to do this `setdefault` business, and can just do `asset_partitions_to_request_map[x].add(y).`

Reply:
ah, i forgot about the pyright...